Tests/TuistPluginTests/PluginServiceTests.swift

```diff
@@ -0,0 +1,134 @@
+            .init(name: "MockPlugin", path: AbsolutePath(pluginPath).appending(component: Constants.helpersDirectoryName))
+        ])
+
+        XCTAssertEqual(plugins, expectedPlugins)
+    }
+
+    func test_loadPlugins_usingConfig_WHEN_localHelpers() throws {
+        let pluginPath = "/path/to/plugin"
+        let config = mockConfig(plugins: [.local(path: pluginPath)])
+
+        mockModelLoader.mockPlugin(pluginPath) { _ in
+            Plugin(name: "MockPlugin")
+        }
+        fileHandler.stubExists = { _ in
+            return true
+        }
+
+        let plugins = try subject.loadPlugins(using: config)
+        let expectedPlugins = Plugins(projectDescriptionHelpers: [
+            .init(name: "MockPlugin", path: AbsolutePath(pluginPath).appending(component: Constants.helpersDirectoryName))
+        ])
+
+        XCTAssertEqual(plugins, expectedPlugins)
+    }
+
+    func test_loadPlugin_atPath_WHEN_gitHelpers() throws {
+        let pluginGitUrl = "https://url/to/repo.git"
+        let pluginGitId = "main"
+        let pluginFingerprint = "\(pluginGitUrl)-\(pluginGitId)".md5
+        let cachedPluginPath = environment.cacheDirectory.appending(components: Constants.PluginDirectory.name, pluginFingerprint)
+
+        mockModelLoader.mockConfig("Config.swift") { _ in
+            self.mockConfig(plugins: [.gitWithBranch(url: pluginGitUrl, branch: pluginGitId)])
+        }
+        mockModelLoader.mockPlugin(cachedPluginPath.pathString) { _ in
+            Plugin(name: "MockPlugin")
+        }
+        fileHandler.stubExists = { _ in
+            return true
+        }
+
+        let plugins = try subject.loadPlugins(at: try temporaryPath())
+        let expectedPlugins = Plugins(projectDescriptionHelpers: [
+            .init(name: "MockPlugin", path: cachedPluginPath.appending(component: Constants.helpersDirectoryName))
+        ])
+
+        XCTAssertEqual(plugins, expectedPlugins)
+    }
+
+    func test_loadPlugin_usingConfig_WHEN_gitHelpers() throws {
+        let pluginGitUrl = "https://url/to/repo.git"
+        let pluginGitId = "main"
+        let pluginFingerprint = "\(pluginGitUrl)-\(pluginGitId)".md5
+        let cachedPluginPath = environment.cacheDirectory.appending(components: Constants.PluginDirectory.name, pluginFingerprint)
+        let config = mockConfig(plugins: [.gitWithBranch(url: pluginGitUrl, branch: pluginGitId)])
+
+        mockModelLoader.mockPlugin(cachedPluginPath.pathString) { _ in
+            Plugin(name: "MockPlugin")
+        }
+        fileHandler.stubExists = { _ in
+            return true
+        }
+
+        let plugins = try subject.loadPlugins(using: config)
+        let expectedPlugins = Plugins(projectDescriptionHelpers: [
+            .init(name: "MockPlugin", path: cachedPluginPath.appending(component: Constants.helpersDirectoryName))
+        ])
+
+        XCTAssertEqual(plugins, expectedPlugins)
+    }
+
+    private func mockConfig(plugins: [PluginLocation]) -> Config {
```

Comment:
Is there a reason you are not using `Config.test()` extension function defined in `Config+TestData.swift` ?

Reply:
Ah right we I should totally use that one instead!